tests/unit/test_parallel.py

```diff
@@ -117,3 +118,38 @@ def works():
     assert [True, True, True, True] == results
     assert 0 == len(errors)
     assert ["Finished 'testing' tasks: 100% results available (4/4)"] == _predictable_messages(caplog)
+
+
+def test_odd_partial_failed(caplog):
+    caplog.set_level(logging.INFO)
+
+    def fails_on_odd(n=1, dummy=None):
+        if isinstance(n, str):
+            raise RuntimeError("strings are not supported!")
+
+        if n % 2:
+            msg = "failed"
+            raise DatabricksError(msg)
+
+    tasks = [
+        partial(fails_on_odd, n=1),
+        partial(fails_on_odd, 1, dummy="6"),
+        partial(fails_on_odd),
+        partial(fails_on_odd, n="aaa"),
+    ]
+
+    signatures = [Threads._get_result_function_signature(func, "test") for func in tasks]
```

Comment:
don't call private methods in the test. this is going to be detected by a custom pylint check.